// @cloudflare/ai-chat/dist/types.js
//#region src/types.ts
/**
* Enum for message types to improve type safety and maintainability
*/
let MessageType = /* @__PURE__ */ function(MessageType$1) {
	MessageType$1["CF_AGENT_CHAT_MESSAGES"] = "cf_agent_chat_messages";
	MessageType$1["CF_AGENT_USE_CHAT_REQUEST"] = "cf_agent_use_chat_request";
	MessageType$1["CF_AGENT_USE_CHAT_RESPONSE"] = "cf_agent_use_chat_response";
	MessageType$1["CF_AGENT_CHAT_CLEAR"] = "cf_agent_chat_clear";
	MessageType$1["CF_AGENT_CHAT_REQUEST_CANCEL"] = "cf_agent_chat_request_cancel";
	/** Sent by server when client connects and there's an active stream to resume */
	MessageType$1["CF_AGENT_STREAM_RESUMING"] = "cf_agent_stream_resuming";
	/** Sent by client to acknowledge stream resuming notification and request chunks */
	MessageType$1["CF_AGENT_STREAM_RESUME_ACK"] = "cf_agent_stream_resume_ack";
	/** Client sends tool result to server (for client-side tools) */
	MessageType$1["CF_AGENT_TOOL_RESULT"] = "cf_agent_tool_result";
	/** Server notifies client that a message was updated (e.g., tool result applied) */
	MessageType$1["CF_AGENT_MESSAGE_UPDATED"] = "cf_agent_message_updated";
	return MessageType$1;
}({});

//#endregion
export { MessageType };
